AndroidNativeKotlinTemplate/app/src/main/java/com/salesforce/androidnativekotlintemplate/QrCodeEnabledLoginActivity.kt

```diff
@@ -35,6 +43,29 @@ class QrCodeEnabledLoginActivity : LoginActivity() {
 
         // Find and show the layout's QR code login button.
         findViewById<Button>(qr_code_login_button).visibility = VISIBLE
+
+        // Create and register a broadcast intent receiver to finish this activity on successful authentication.
+        finishOnUserSwitchBroadcastReceiver = object : BroadcastReceiver() {
+            override fun onReceive(context: Context?, intent: Intent) {
+                /*
+                 * Note: When this activity starts the intent for main activity
+                 * after scanning the QR code, finishing prior to log in
+                 * completion would also finish the activities being used for
+                 * the in-progress log in.
+                 *
+                 * If the main activity intent could be started in such a way
+                 * that this activity could finish immediately after starting
+                 * the intent, this receiver wouldn't be needed.
+                 */
+                if (intent.action == USER_SWITCH_INTENT_ACTION) finish()
```

Comment:
@brandonpage or @wmathurin, here's a question that should be easy to solve, but for some reason I was stumped.

A little later on, this activity wants to start an intent for main activity with the QR Code Log In data.  That's fine, except if this intent finishes before the user is authenticated, it finishes the entire app back to the launcher.

I tried quite a few different permutations of setting up the intent and activities, but to no avail.  Since it finishes with no consequence _after_ the user is authenticated I'm reckoning there's logic inherited from LoginActivity that is up to something.  With this work-around, perhaps that's not a release blocker, but I really do just want to know what it is.  🧐

Reply:
Seems similar to this situation maybe? https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/c998f718172710896e724706875f5e4a28a7c891/libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/NativeLoginManager.kt#L228

Reply:
I'm not sure that's the same issue, but I _did_ find the root cause.  `LoginActivity` has some logic where it's making some private choices about the tasks and back stack for us.  In this case, there's a trigger where it's sending the task with it and `MainActivity` to the back too soon.

https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/c998f718172710896e724706875f5e4a28a7c891/libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.kt#L432

The interesting thing is I'm trying to test all the cases the comment mentioned - they all seem to work just as well with only `finish()` instead of `moveTaskToBack(true)`.  In this event we're actually coming in from `onDestroy()` since the activity is already finished.

I did some research into this logic and it's been in place since this was Java in 2011.
<img width="1067" alt="Screenshot 2024-10-10 at 12 25 04" src="https://github.com/user-attachments/assets/9a2e1315-32ee-47a3-ae01-8ef13df26ef9">

@brandonpage or @wmathurin - With all the changes that have been made over the years to the way `LoginActivity` is placed on the task and back stack, is this even needed?

To minimize risk, we could certainly keep this work-around in the template for a moment.  Then we could analyze this in more detail for a fix or fully understanding it.

Reply:
@JohnsonEricAtSalesforce It's possible that it is no longer needed on newer devices but older devices may act differently.  For example if you tested on Android 13+ it would use the completely new [predictive back system](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2617/files).  I would also be cautious because LoginActivty is sometimes a lock screen (Bio Auth) and the behavior around custom tabs can further complicate things since it launchers out LoginActivity.

Reply:
Thanks for the feedback plus the history, @brandonpage. I'm thinking we minimize risk for now and not change this.  In light of the launchMode changes maybe we can revisit this as a maintenance topic soon after the next release.